### PR TITLE
feat: add budoux-aware iW/aW text objects

### PR DESCRIPTION
**Pull request by atusy**
## Summary

- Adds `iW`/`aW` text objects where a Japanese WORD is a budoux segment: `require("budouxify.textobject").find()` returns the range, `select()` drives visual/operator-pending keymaps (README shows the mappings).
- Vim-conformant behavior: blanks-run `iW`, trailing/leading-blank rules for `aW` (line-leading indentation stays), counts (`d2aW`), `selection=exclusive`, past-EOL cursors; deliberate differences (single-line, visual reselect) are documented.
- Extracts the WORD tokenizer into `budouxify.words` shared by text objects and B/gE, mirrors Neovim's structural UTF-8 stepping, and hardens it against NUL, LF, control chars, invalid UTF-8 (former infinite loops), and Vim's full Unicode blank table (NBSP, U+1680, U+2000-200B, ...).
- Fixes pre-existing bugs found along the way: `_parse_words` hang on invalid UTF-8, `find_forward` raising instead of reporting cursor errors, budoux model reloaded per call, README install snippet syntax.

## Test plan

- `make test` (mini.test): 220+ cases including a visual byte-offset DSL, operator-level `diW`/`daW`/`viW` runs, count/virtualedit/selection=exclusive pins, and tokenizer edge cases.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by atusy** [comment]
@coderabbitai review
@greptileai
@codex review
/agentic_review

File: lua/budouxify/textobject.lua [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	end
+	return range
+end
+
+--- Visually select the range for use in operator-pending and visual mode mappings.
+---@param opts { inner: boolean, error_handler?: function }
+function M.select(opts)
+	local range = M.find({ inner = opts.inner, count = vim.v.count1, error_handler = opts.error_handler })
+	if not range then
+		return
```
[the remark this answers is not in the document]
Comment (atusy) [reply lua/budouxify/textobject.lua]: Not reproducible — the pending operator does not survive a no-op motion. Verified with feedkeys('diW', 'x') on an empty line with this mapping: vim.fn.mode(1) is back to "n" immediately afterwards, and a following j just moves the cursor (buffer intact). Vim aborts the operator when the o-mode mapping ends without a selection. Pinned as a regression test (diWj on an empty line leaves the buffer unchanged) in 6f5a5ea.

**Comment by atusy** [comment]
@codex review
